cli/src/cmd_net.rs

```diff
@@ -420,6 +434,275 @@ impl AuthenticatedCmd for CmdBgpFilter {
+                    .find(|x| x.addr == self.peer)
+                    .ok_or(anyhow::anyhow!("specified peer does not exist"))?;
+
+                peer.md5_auth_key = self.authstring.clone();
+            }
+        }
+        client
+            .networking_switch_port_settings_create()
+            .body(settings)
+            .send()
+            .await?;
+        Ok(())
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "pref")]
+pub struct CmdBgpLocalPref {
+    /// Id of the rack to add the route preference to.
+    #[arg(long)]
+    rack: Uuid,
+
+    /// Switch to add the route preference to.
+    #[arg(long, value_enum)]
+    switch: Switch,
+
+    /// Port to add the route preference to.
+    #[arg(long, value_enum)]
+    port: Port,
+
+    /// Peer to add the route preference to.
+    #[arg(long)]
+    peer: IpAddr,
+
+    /// Local preference.
+    #[clap(long)]
+    local_pref: Option<u32>,
+}
+
+#[async_trait]
+impl AuthenticatedCmd for CmdBgpLocalPref {
+    async fn run(&self, client: &Client) -> Result<()> {
+        let mut settings =
+            current_port_settings(client, &self.rack, &self.switch, &self.port).await?;
+        match settings.bgp_peers.get_mut(PHY0) {
+            None => return Err(anyhow::anyhow!("no BGP peers configured")),
+            Some(config) => {
+                let peer = config
+                    .peers
+                    .iter_mut()
+                    .find(|x| x.addr == self.peer)
+                    .ok_or(anyhow::anyhow!("specified peer does not exist"))?;
+
+                peer.local_pref = self.local_pref
+            }
+        }
+        client
+            .networking_switch_port_settings_create()
+            .body(settings)
+            .send()
+            .await?;
+        Ok(())
+    }
+}
+
+/// Manage static switch routes.
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "route")]
+pub struct CmdStaticRoute {
+    #[clap(subcommand)]
+    subcommand: RouteSubCommand,
+}
+
+#[async_trait]
+impl AuthenticatedCmd for CmdStaticRoute {
+    async fn run(&self, client: &Client) -> Result<()> {
+        match &self.subcommand {
+            RouteSubCommand::Set(cmd) => cmd.run(client).await,
+            RouteSubCommand::Delete(cmd) => cmd.run(client).await,
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+enum RouteSubCommand {
+    /// Set a static route.
+    Set(CmdStaticRouteSet),
+    /// Delete a static route.
+    Delete(CmdStaticRouteDelete),
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "route set")]
+pub struct CmdStaticRouteSet {
+    /// Id of the rack to configure the route on.
+    #[arg(long)]
+    rack: Uuid,
+
+    /// Switch to configure the route on.
+    #[arg(long, value_enum)]
+    switch: Switch,
+
+    /// Port to configure the route on.
+    #[arg(long, value_enum)]
+    port: Port,
+
+    /// The route destination.
```

Comment:
See comment for line 570, the help strings for `CmdStaticRouteDelete` are the same as here and below.

Reply:
See response to line 570 comment